internal/services/mobilenetwork/mobile_network_attached_data_network_resource.go

```diff
@@ -0,0 +1,617 @@
+			}
+
+			if metadata.ResourceData.HasChange("user_equipment_static_address_pool_prefixes") {
+				addtachedDataNetwork.Properties.UserEquipmentStaticAddressPoolPrefix = &plan.UserEquipmentStaticAddressPoolPrefix
+			}
+
+			if metadata.ResourceData.HasChange("user_plane_access_name") {
+				addtachedDataNetwork.Properties.UserPlaneDataInterface.Name = &plan.UserPlaneAccessName
+			}
+
+			if metadata.ResourceData.HasChange("user_plane_access_ipv4_address") {
+				addtachedDataNetwork.Properties.UserPlaneDataInterface.IPv4Address = &plan.UserPlaneAccessIPv4Address
+			}
+
+			if metadata.ResourceData.HasChange("user_plane_access_ipv4_subnet") {
+				addtachedDataNetwork.Properties.UserPlaneDataInterface.IPv4Subnet = &plan.UserPlaneAccessIPv4Subnet
+			}
+
+			if metadata.ResourceData.HasChange("user_plane_access_ipv4_gateway") {
+				addtachedDataNetwork.Properties.UserPlaneDataInterface.IPv4Gateway = &plan.UserPlaneAccessIPv4Gateway
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				addtachedDataNetwork.Tags = &plan.Tags
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, *addtachedDataNetwork); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r AttachedDataNetworkResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MobileNetwork.AttachedDataNetworkClient
+
+			id, err := attacheddatanetwork.ParseAttachedDataNetworkID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := AttachedDataNetworkModel{
+				Name:                               id.AttachedDataNetworkName,
+				MobileNetworkPacketCoreDataPlaneId: packetcoredataplane.NewPacketCoreDataPlaneID(id.SubscriptionId, id.ResourceGroupName, id.PacketCoreControlPlaneName, id.PacketCoreDataPlaneName).ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				props := model.Properties
+
+				state.Location = location.Normalize(model.Location)
+				state.DnsAddresses = props.DnsAddresses
+				state.NaptConfiguration = flattenNaptConfigurationModel(props.NaptConfiguration)
+				state.UserEquipmentAddressPoolPrefix = pointer.From(props.UserEquipmentAddressPoolPrefix)
+				state.UserEquipmentStaticAddressPoolPrefix = pointer.From(props.UserEquipmentStaticAddressPoolPrefix)
+				state.UserPlaneAccessIPv4Address = pointer.From(props.UserPlaneDataInterface.IPv4Address)
+				state.UserPlaneAccessIPv4Gateway = pointer.From(props.UserPlaneDataInterface.IPv4Gateway)
+				state.UserPlaneAccessIPv4Subnet = pointer.From(props.UserPlaneDataInterface.IPv4Subnet)
+				state.UserPlaneAccessName = pointer.From(props.UserPlaneDataInterface.Name)
```

Comment:
Can `props.UserPlaneDataInterface` be nil?

Reply:
Nope